apps/browser/src/vault/fido2/content/messaging/messenger.ts

```
@@ -127,4 +101,83 @@ export class Messenger {
       localPort.close();
     }
   }
+
+  private createMessageEventListener() {
+    return async (event: MessageEvent<MessageWithMetadata>) => {
+      const windowOrigin = window.location.origin;
+      if (event.origin !== windowOrigin) {
+        return;
+      }
+
+      if (this.handler === undefined) {
+        return;
+      }
+
+      const message = event.data;
+      const port = event.ports?.[0];
+      if (
+        message?.SENDER !== SENDER ||
+        message.senderId == this.messengerId ||
+        message == null ||
+        port == null
+      ) {
+        return;
+      }
+
+      const abortController = new AbortController();
+      port.onmessage = (event: MessageEvent<MessageWithMetadata>) => {
+        if (event.data.type === MessageType.AbortRequest) {
+          abortController.abort();
+        }
+      };
+
+      let onDestroyListener;
+      const destroyPromise: Promise<never> = new Promise((_, reject) => {
+        onDestroyListener = () => reject(new Error("Extension is being destroyed"));
```

suggestion: Maybe return a fallback error instead? I think users are gonna be very confused if they get the error `Extension is being destroyed`